src/resolver/const_evaluator.rs

```diff
@@ -642,7 +642,25 @@ fn get_cast_statement_literal(
             }
         }
 
-        //Some(&crate::typesystem::DataTypeInformation::Float{..}) => {},
+        Some(DataTypeInformation::Float { .. }) => {
+            let evaluated = evaluate(cast_statement, scope, index)?;
+            let value = match evaluated {
+                Some(AstStatement::LiteralInteger { value, .. }) => Some(value as f64),
+                Some(AstStatement::LiteralReal { value, .. }) => value.parse::<f64>().ok(),
+                _ => return Err(format!("Expected floating point type, got: {evaluated:?}")),
```

Comment:
Could you add tests address the codecov complaints? Casting a string as REAL should result in an error